astropy/config/configuration.py

```diff
@@ -570,6 +574,57 @@ def get_config(packageormod=None, reload=False, rootname=None):
         return cobj
 
 
+def generate_config(filename=None):
+    """Generates a configuration file, from the list of ConfigItem
+    for each subpackage.
+    """
+
+    # First, we need to import all modules containing a ConfigNamespace
+    # subclass. So this list must be kept up-to-date.
+    modules_to_import = [
+        'astropy',
+        'astropy.io.fits',
+        'astropy.io.votable',
+        'astropy.logger',
+        'astropy.nddata',
+        'astropy.samp',
+        'astropy.table',
+        'astropy.table.jsviewer',
+        'astropy.units.quantity',
+        'astropy.utils.data',
+        'astropy.utils.iers.iers',
+        'astropy.visualization.wcsaxes',
+    ]
+    for mod in modules_to_import:
+        importlib.import_module(mod)
+
+    wrapper = TextWrapper(initial_indent="## ", subsequent_indent='## ',
+                          width=78)
+
+    fp = open(filename, 'w') if filename else sys.stdout
```

Comment:
Just a personal opinion: I find `sys.stdout` not useful except for debugging. A more useful feature is to return the stuff as one string (with line breaks), a list of strings (without line breaks), or dictionary. Then user can do whatever user wants with that output (write to file, print to screen, etc). Also easier to test.

Reply:
The idea is that stdout may be useful for docs inclusion, but we will see. Otherwise I think it would be useful to save this config file in its default location, and print where this is.

Reply:
I would prefer using `.. literalinclude::` over `stdout`.

Reply:
I think @pllim is right - better to just have the ability to get a list of strings, or an `StringIO()` object, even for creating a documentation page.

Reply:
Ok, I changed to accept file objects instead.